### PR TITLE
Support maven revision

### PR DESCRIPTION
**Pull request by zhaoyunxing92**
### <Feature description>
- [x] [Using of CI friendly versions ](https://maven.apache.org/docs/3.5.0/release-notes.html#overview-about-the-changes) eg: `mvnd clean package -Dmaven.test.skip=true -Drevision=8.16.12 -Pall`

### CODE REVIEW

**Comment by wu-sheng** [comment]
This is never discussed. Why do you add this?

**Comment by zhaoyunxing92** [comment]
> This is never discussed. Why do you add this?

Unify the version management in the parent class, which is convenient for publishing and packaging without modifying a version file every time you publish

**Comment by lujiajing1126** [comment]
> > This is never discussed. Why do you add this?
> 
> Unify the version management in the parent class, which is convenient for publishing and packaging without modifying a version file every time you publish

Is this compatible with the maven-publish plugin as well as the Apache release procedure?

**Comment by zhaoyunxing92** [comment]
> > > This is never discussed. Why do you add this?
> > 
> > 
> > Unify the version management in the parent class, which is convenient for publishing and packaging without modifying a version file every time you publish
>
> Is this compatible with the maven-publish plugin as well as the Apache release procedure?

Compatible. In the past, `mvn versions:set -DnewVersion=9.1.0` may be used to modify the version number in batches. Now it is only used to modify one place in the pom file. If you use the command, you can use `-Drevision=9.1.0` to modify, eg: `mvnd clean package -Dmaven.test.skip=true -Drevision=8.16.12 -Pall`

**Comment by wu-sheng** [comment]
I can't see the point of using this. If you need to package for specific version, why don't you use the give git tag?

**Comment by kezhenxu94** [comment]
> > > > This is never discussed. Why do you add this?
> > > 
> > > 
> > > Unify the version management in the parent class, which is convenient for publishing and packaging without modifying a version file every time you publish
> > 
> > 
> > Is this compatible with the maven-publish plugin as well as the Apache release procedure?
> 
> Compatible. In the past, `mvn versions:set -DnewVersion=9.1.0` may be used to modify the version number in batches. Now it is only used to modify one place in the pom file. If you use the command, you can use `-Drevision=9.1.0` to modify, eg: `mvnd clean package -Dmaven.test.skip=true -Drevision=8.16.12 -Pall`

Check the full steps about how to do the release, and verify the changes in this PR is OK in those process

https://github.com/apache/skywalking/blob/1256f7059df3affd73c27d1b17a3cf8d7ea6efff/docs/en/guides/How-to-release.md?plain=1#L48

If so I'm glad to approve the changes so that we don't have to change hundreds of lines in every release.

**Comment by zhaoyunxing92** [comment]
> > > > > This is never discussed. Why do you add this?
> > > > 
> > > > 
> > > > Unify the version management in the parent class, which is convenient for publishing and packaging without modifying a version file every time you publish
> > > 
> > > 
> > > Is this compatible with the maven-publish plugin as well as the Apache release procedure?
> > 
> > 
> > Compatible. In the past, `mvn versions:set -DnewVersion=9.1.0` may be used to modify the version number in batches. Now it is only used to modify one place in the pom file. If you use the command, you can use `-Drevision=9.1.0` to modify, eg: `mvnd clean package -Dmaven.test.skip=true -Drevision=8.16.12 -Pall`
> 
> That's not how we do a release, check the full steps about how to do the release, and verify the changes in this PR is OK in those process
> 
> https://github.com/apache/skywalking/blob/1256f7059df3affd73c27d1b17a3cf8d7ea6efff/docs/en/guides/How-to-release.md?plain=1#L48
> 
> If so I'm glad to approve the changes so that we don't have to change hundreds of lines in every release.

Yes, you only need to modify one file for future releases

**Comment by wu-sheng** [comment]
We don't need to change any files manually. We just run one command only. That is my confusion.

**Comment by zhaoyunxing92** [comment]
But the modified files are different, now only one file needs to be modified, even without modifying the command

**Comment by zhaoyunxing92** [comment]
I saw that a bunch of files have to be modified every time I publish. Now I don’t need to modify the files, or I only need to modify one, and this is compatible with your previous ones. This is supported in maven3.5.0

**Comment by zhaoyunxing92** [comment]
> You are not reading clearly about Zhenxu posted previously. [This command](https://github.com/apache/skywalking/blob/1256f7059df3affd73c27d1b17a3cf8d7ea6efff/docs/en/guides/How-to-release.md?plain=1#L48) would do all the changes and even `git tag`/`git commit` from v1-SNAPSHOT to v1 and then v2-SNAPSHOT automatically.

I have read this, but I feel that I can try a new release method. I modified too many files in each previous release, and now I don’t need to modify the files.

**Comment by wu-sheng** [comment]
That is your practice, not an upstream case. 
For us, you are actually adding a new manual file change, rather than an all automatic process.

**Comment by zhaoyunxing92** [comment]
> That is your practice, not an upstream case. For us, you are actually adding a new manual file change, rather than an all automatic process.

He is compatible with the previous release method

File: pom.xml [hunk 1]
```diff
@@ -79,6 +79,7 @@
     <properties>
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
         <compiler.version>8</compiler.version>
+        <revision>9.1.0-SNAPSHOT</revision>
```
Comment (wu-sheng) [inline pom.xml:82]: This has to be manually changed to 9.1.0 for release and tag, and then 9.2.0-SNAPSHOT for the next dev stage. 
What do you mean as same as before? You are not an upstream release manager, I don't think you have gone through all Java agent release process.

Comment (kezhenxu94) [reply pom.xml:82]: > This has to be manually changed to 9.1.0 for release and tag, and then 9.2.0-SNAPSHOT for the next dev stage.

This doesn't actually need to be manually changed, when you run the release command, you'll be prompted like before about what the version is, what the next SNAPSHOT version is

**Pull request closed without merging**